framework/crud/eager.go

```
@@ -165,9 +171,14 @@ func eagerLoadHasMany(ctx context.Context, db DBExecutor, safeEntity, safeFK str
 	}
 
 	scopeClause, scopeArgs := filterClause(scopeFilters, len(ids)+1)
-	q := fmt.Sprintf("SELECT * FROM %s WHERE %s IN (%s)%s%s",
-		query.QuoteIdent(safeEntity), query.QuoteIdent(safeFK), strings.Join(placeholders, ", "), softDeleteFilter, scopeClause)
+	readClause, readArgs := renderReadScope(readPreds, "", len(ids)+len(scopeArgs)+1)
+	if readClause != "" {
+		readClause = " AND " + readClause
+	}
+	q := fmt.Sprintf("SELECT * FROM %s WHERE %s IN (%s)%s%s%s",
+		query.QuoteIdent(safeEntity), query.QuoteIdent(safeFK), strings.Join(placeholders, ", "), softDeleteFilter, scopeClause, readClause)
 	args = append(args, scopeArgs...)
```

golangci-lint (2.12.2)

[error] append to slice `args` with non-zero initialized length

(makezero)